tool/check-code.sh

```diff
@@ -87,86 +84,62 @@ function check_formatting() {
   fi
 }
 
+# Extract snippets, analyze and check formatting
+# Null safety flag set
+EXAMPLE_ROOT="examples"
+echo "=> ANALYZING and testing apps in $EXAMPLE_ROOT/* --null-safety"
+
+for sample in $EXAMPLE_ROOT/*/*{,/*}; do
+
+  # Does it have a pubspec?
+  if [[ -d "$sample" && -e "$sample/pubspec.yaml" ]]; then
```

Comment:
Looks like GitHub Actions has 2-core CPUs. Could we try to run 2 of these sample tests in parallel at the same time? Something along the lines of.

test Sample1 &
test sample2 &
wait

Reply:
Yes, absolutely - this would be the place to run these these on multiple threads. I suggest we get this new baseline in place, evaluate, then come back in and test these additional optimizations.

Reply:
Got it! You mentioned did that in the email, we can loop back on this one!

Reply:
Another followup optimization might be not to use the local Docker container at all for tests, since it now _only_ uses Flutter. Using a 3rd party action might just be faster and worth testing for the CI side of things, and the container is great for local testing and completeness:

```
steps:
      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
         with:
            submodules: recursive
      - uses: subosito/flutter-action@v2
         with:
            channel: ${{ matrix.channel }}
      - name: Test
         run: tool/test.sh --target ${{ matrix.channel }}
         continue-on-error: ${{ matrix.experimental }}
```

Reply:
the third party action would be good to look at after we parallelize the tests. Looks like the tests are responsible for majority of the build time, so breaking it out might only save us a few minutes?